pkg/controller/replicaset/metrics/metrics.go

```diff
@@ -0,0 +1,43 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metrics
+
+import (
+	"sync"
+
+	"k8s.io/component-base/metrics"
+	"k8s.io/component-base/metrics/legacyregistry"
+)
+
+const ReplicaSetControllerSubsystem = "replicaset_controller"
+
+var SortingDeletionAgeRatio = metrics.NewHistogram(
+	&metrics.HistogramOpts{
+		Subsystem: ReplicaSetControllerSubsystem,
+		Name:      "sorting_deletion_age_ratio",
+		Help:      "The ratio of chosen deleted pod's ages to the current youngest pod's age (at the time). Should be <2",
+	},
+)
+
+var registerMetrics sync.Once
+
+// Register registers ReplicaSet controller metrics.
+func Register() {
```

Comment:
Can we pass the registry as an argument here in order to prepare for an eventual decoupling of the global registry at some point? An example can be found here:

Register function: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/kube-aggregator/pkg/controllers/status/metrics.go#L64-L80

Registration: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/kube-aggregator/pkg/controllers/status/available_controller.go#L214-L221

Reply:
Does this need the `customRegistrationFunc` parameter too, with a `collector`? Or is it sufficient to just pass the registration func?

Reply:
The registration function is sufficient here, you don't need to pass a custom one.